pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/pulsar/client/kafka/compat/KafkaProducerInterceptorWrapper.java

```diff
@@ -66,9 +66,9 @@
     final private org.apache.kafka.clients.producer.ProducerInterceptor<K, V> kafkaProducerInterceptor;
 
     // For serializer key/value, and to determine the deserializer for key/value.
-    private final Serializer<K> keySerializer;
+    private final PulsarKafkaSchema<K> keySerializer;
```

Comment:
Why can't you use `Schema<K>`?

Reply:
As above, I understand that this KafkaProducerInterceptorWrapper  class should be specially prepared for kafka, in this class, there are several logics that will use the implementation class PulsarKafkaSchema. If defined as Schema, cast type convert is required everywhere. 
Just like the following code:
```
(PulsarKafkaSchema<K>) keySerializer.setTopic(topic);
(PulsarKafkaSchema<K>) keySerializer.getKafkaSerializer()
(PulsarKafkaSchema<V>) valueSerializer.setTopic(topic);
(PulsarKafkaSchema<V>) valueSerializer.getKafkaSerializer()
```

Reply:
I don't understand your point. If user passes a Pulsar schema in, the user should still be able to use ProducerInterceptor, right?

Reply:
I have updated the code. Please help me review it again. I am still not sure whether I need to implement a new intercept under the current situation for Pulsar schema. Use the following code:
```
pulsarProducerBuilder.clone().topic(topic).intercept(pulsarProducerInterceptor).create();
```

Reply:
@tuteng my comment is not asking you to implement a new interceptor. my comment is why do you only add interceptors when the schema is a PulsarKafkaSchema. The interceptors should be working no matter if the schema is a PulsarKafkaSchema or a normal Pulsar schema, right?

```
if (keySerializer instanceof PulsarKafkaSchema && valueSerializer instanceof PulsarKafkaSchema) {
```

Reply:
done